server/channels/db/migrations/postgres/000115_user_reporting_changes.up.sql

```diff
@@ -0,0 +1,7 @@
+ALTER TABLE users ADD COLUMN IF NOT EXISTS lastlogin bigint NOT NULL DEFAULT 0;
+
+CREATE MATERIALIZED VIEW IF NOT EXISTS poststats AS
+SELECT UserId, FLOOR(CreateAt / 86400000) AS Day, COUNT(*) as NumPosts, MAX(CreateAt) as LastPostDate
```

Comment:
The `FLOOR(CreateAt / 86400000)` is giving incorrect results into the future, and also we need to re-multiply with `100000` to get back into ms. I'd suggest to skip the epoch format and directly use the postgres timestamp format for better usability. Suggest `to_timestamp(createat/1000)::date as day` instead.

Also, since this code was already buggy, let's add a unit test for this that fills up a posts table, then creates the materialized view and ensures the results are correct.

Reply:
First part done. As for the unit test, I think i'll create one once we have the endpoint set up that will actually select from this table, as with the current PR that currently doesn't exist.

So this will be tested in a follow-up PR.

Reply:
cool